README.md

```diff
@@ -1,9 +1,102 @@
+==================
+### A Symfony Bundle for easy configuration management.
 
+## Features Include
+* Group wise dynamic form creation ability
+* Store global configuration
+* Store user wise configuration
+* Get configuration through rest API
 
+## Install
+1. Download EasyConfigBundle using composer
+2. Enable the bundle
+3. Register an Interface to Kernel file
+4. Create Config entity class
+5. Configure yaml file
+6. Update database schema 
+7. Create your first form 
+8. Register the bundle’s routes
+9. Overriding default EasyConfigBundle templates (optional)
 
-### License
 
+### 1. Download EasyConfigBundle using composer :
+Open a command console, enter project directory and execute the following command to download the latest stable version of this bundle:
+```bash
+$ composer require xiidea/easy-config-bundle
+```
+
+### 2. Enable the Bundle:
+The bundle should be automatically enabled by Symfony Flex. If you don't use [Flex](https://symfony.com/doc/current/setup/flex.html), you will need to enable the bundle manually by adding the following line in the config/bundles.php file of your project:
+
+```php
+<?php
+// src/Kernel.php
+
+return [
+    // ...
+    Xiidea\EasyConfigBundle\XiideaEasyConfigBundle::class => ['all' => true],
+];
+```
+
+### 3. Register an Interface to Kernel file
+Open application Kernel.php File and add below code inside the build method of this file
+```php
+<?php
+// src/Kernel.php
+
+public function build(ContainerBuilder $container)
+{
+    // ...
+    $container->registerForAutoconfiguration(ConfigGroupInterface::class)
+	   ->addTag('xiidea.easy_config.group');
+}
+```
+
+_Note: Do not forget to include the below line above in your src/Kernel.php_
+
+`use Xiidea\EasyConfigBundle\Services\FormGroup\ConfigGroupInterface;`
+
+### 4. Create Config entity class:
+By default `EasyConfigBundle` supports Doctrine ORM. However, you must provide an Entity class and the class has to extend the class  `\Xiidea\EasyConfigBundle\Model\BaseConfig`. To configure the Entity class properly please follow the detailed [instructions](https://github.com/xiidea/EasyConfigBundle/blob/main/Resources/doc/config-entity.md).
+
+### 5. Configure yaml file:
+Create a file in the following directory `config/packages` with the exact name `xiidea_easy_config.yaml`
+A sample config file is available in this path `Resources/config/config-sample.yml`. Copy this sample file's content and paste to just created file. Do not forget to change your Entity class name which has been mentioned in [Step 4](https://github.com/xiidea/EasyConfigBundle/blob/main/Resources/doc/config-entity.md).
+```yaml
+# Xiidea Easy Config Configuration Sample
+
+xiidea_easy_config:
+    config_class: App\Entity\Configuration
+```
+### 6. Update database schema:
+It’s time to set up the database schema, open your command console, go to your project root path and execute the following command.
+
+```bash
+$ php bin/console doctrine:schema:update --force
```

Comment:
Instead of doctrine scheme update command we may provide table creation schema script

Reply:
Though it's not related to the readme file but rather an implementation; It's best practice for a bundle to provide a `Model` instead of an `Entity` so user can define their own model with their own table name and field name only by extending the Model. you can get ideas from easy-audit-bundle.

Reply:
As we are giving the opportunity to users that they can provide their ORM/ODM with necessary fields, so we will not provide table creation schema here.